gdal/port/cpl_vsil_hdfs.cpp

```diff
@@ -154,18 +154,27 @@ VSIHdfsHandle::Read(void *pBuffer, size_t nSize, size_t nMemb)
   if (nSize == 0 || nMemb == 0)
     return 0;
 
-  int bytes = hdfsRead(poFilesystem, poFile, pBuffer, nSize * nMemb);
+  int bytes_wanted = nSize * nMemb;
```

Comment:
isn't there a potential issue of size_t -> int overflow in case of big input arguments ?

Reply:
The return type of `hdfsRead` seems to be `tSize` (https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs/include/hdfs/hdfs.h#L585-L600) which is a typedef of `int32_t` (https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs/include/hdfs/hdfs.h#L75), so an `int` "should be" sufficient.  Maybe I should change those from `int`s to `int32_t`s just to be safe though?

Reply:
I see what you are saying: you are saying that there is a mismatch between the two last arguments to `Read` (of type `size_t` which I think is `uint64_t`) and between the size argument that `hdfsRead` takes (of type `tSize` which is `int32_t`)?

If there is an overflow, should it throw an error or just return as many records as it can?